tabular/src/autogluon/tabular/models/rf/rf_model.py

```diff
@@ -18,6 +19,121 @@
 logger = logging.getLogger(__name__)
 
 
+# FIXME: DOESNT WORK ON REGRESSION PREDICTIONS
+class RFOnnxPredictor:
+    def __init__(self, obj, model):
+        self.num_classes = obj.num_classes
+        self.model = model
+
+    def predict(self, X):
+        input_name = self.model.get_inputs()[0].name
+        label_name = self.model.get_outputs()[0].name
+        return self.model.run([label_name], {input_name: X})[0]
+
+    def predict_proba(self, X):
+        input_name = self.model.get_inputs()[0].name
+        label_name = self.model.get_outputs()[1].name
+        pred_proba = self.model.run([label_name], {input_name: X})[0]
+        pred_proba = np.array([[r[i] for i in range(self.num_classes)] for r in pred_proba])
+        return pred_proba
+
+
+class RFNativeCompiler:
+    name = 'native'
+    save_in_pkl = True
+
+    @staticmethod
+    def can_compile():
+        return True
+
+    @staticmethod
+    def compile(obj, path: str):
+        from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+        from sklearn.ensemble import ExtraTreesRegressor, ExtraTreesClassifier
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+        if isinstance(obj.model, (RandomForestClassifier, RandomForestRegressor,
+                                  ExtraTreesClassifier, ExtraTreesRegressor)):
+            with open(path + 'model_native.pkl', 'wb') as fp:
+                fp.write(pickle.dumps(obj))
+        else:
+            pkl = None
+            with open(path + 'model_native.pkl', 'rb') as fp:
+                pkl = fp.read()
+            return pickle.loads(pkl).model
+        return obj.model
```

Comment:
Move to the end of the `if` statement right before `else`

Reply:
As suggested, this is now modified into
```python
        if isinstance(obj.model, (RandomForestClassifier, RandomForestRegressor,
                                  ExtraTreesClassifier, ExtraTreesRegressor)):
            with open(path + 'model_native.pkl', 'wb') as fp:
                fp.write(pickle.dumps(obj.model))
            return obj.model
        else:
            return RFNativeCompiler.load(obj=obj, path=path)
```